JavaScript/JavaScriptSDK/Util.ts

```diff
@@ -220,8 +220,14 @@ module Microsoft.ApplicationInsights {
         /**
          * helper method to set userId and sessionId cookie
          */
-        public static setCookie(name, value) {
-            Util.document.cookie = name + "=" + value + ";path=/";
+        public static setCookie(name, value, domain?) {
+            var domainAttrib = "";
+
+            if (domain) {
+                domainAttrib = ";domain=" + domain;
```

Comment:
If the `cookieDomain` is not specified we will use a default domain, the same way as it is currently done and it will work in most cases. 

I wouldn't add any code here to guess the domain.